File: bd_api/src/typeorm/restaurant.ts
import {
  Column,
  Entity,
  Index,
  ManyToOne,
  PrimaryGeneratedColumn,
  CreateDateColumn,
  UpdateDateColumn,
  OneToMany,
  OneToOne,
  JoinColumn,
} from 'typeorm';
import { Company } from './company';
import { FileDocument, RestaurantContact } from '.';
import { FileDocumentDto } from 'src/dto/file.dto';

@Entity()
@Index(['name', 'companyId'], { unique: true, nullFiltered: true })
export class Restaurant {
  @PrimaryGeneratedColumn()
  id: number;
  @Column('varchar')
  name: string;
  @Column('varchar', { default: 'no Description' })
  description: string;
  @Column('varchar', { length: 30 })
  email: string;
  @Column('int', { nullable: true })
  companyId: number;
  @Column('varchar', { nullable: true })
  address: string;
  @Column('varchar', { length: 100, nullable: true })
  city: string;
  @Column('varchar', { length: 100, nullable: true })
  country: string;
  @Column('varchar', { length: 20, nullable: true })
  postal_code: string;
  @Column('varchar', { length: 20, nullable: true })
  phone: string;
  @CreateDateColumn()
  createdAt: Date;
  @UpdateDateColumn()
  updatedAt: Date;
  @Column('float', { default: 0.0 })
  laltitude: number;
  @Column('float', { default: 0.0 })
  longitude: number;
  @Column('bool', { default: false })
  isDelecetd: boolean;
  @OneToOne(() => FileDocument, {
    cascade: true,
    onUpdate: 'NO ACTION',
    nullable: true,
  })
  @JoinColumn()
  profile: FileDocument;
  @ManyToOne(() => Company, (company) => company.restaurants, {
    onDelete: 'SET NULL',
    onUpdate: 'SET NULL',
    nullable: true,
  })
  company: Company;
  @OneToMany(() => RestaurantContact, (rest) => rest.restaurant)
  contact: RestaurantContact;
  @Column({ type: 'time', default: '08:00' })
  openingTime: string;

  @Column({ type: 'time', default: '23:00' })
  closingTime: string;
}

File: mobile/lib/cores/model/restaurant.dart
import 'package:mobile/cores/model/company.dart';
import 'package:mobile/cores/model/document.dart';

class Restaurant {
  int? id;
  String? name;
  String? email;
  int? companyId;
  String? address;
  String? city;
  String? country;
  String? postalCode;
  String? phone;
  String? createdAt;
  String? updatedAt;
  double? laltitude;
  double? longitude;
  bool? isDelecetd;
  String? openingTime;
  String? closingTime;
  Company? company;
  Photo? profile;

  Restaurant(
      {this.id,
      this.name,
      this.email,
      this.companyId,
      this.address,
      this.city,
      this.country,
      this.postalCode,
      this.phone,
      this.createdAt,
      this.updatedAt,
      this.laltitude,
      this.longitude,
      this.isDelecetd,
      this.openingTime,
      this.closingTime,
      this.company,
      this.profile});

  Restaurant.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    email = json['email'];
    companyId = json['companyId'];
    address = json['address'];
    city = json['city'];
    country = json['country'];
    postalCode = json['postal_code'];
    phone = json['phone'];
    createdAt = json['createdAt'];
    updatedAt = json['updatedAt'];
    laltitude = double.parse(json['laltitude'].toString());
    longitude = double.parse(json['longitude'].toString());
    isDelecetd = json['isDelecetd'];
    openingTime = json['openingTime'];
    closingTime = json['closingTime'];
    company =
        json['company'] != null ? new Company.fromJson(json['company']) : null;
    profile =
        json['profile'] != null ? new Photo.fromJson(json['profile']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['name'] = this.name;
    data['email'] = this.email;
    data['companyId'] = this.companyId;
    data['address'] = this.address;
    data['city'] = this.city;
    data['country'] = this.country;
    data['postal_code'] = this.postalCode;
    data['phone'] = this.phone;
    data['createdAt'] = this.createdAt;
    data['updatedAt'] = this.updatedAt;
    data['laltitude'] = this.laltitude;
    data['longitude'] = this.longitude;
    data['isDelecetd'] = this.isDelecetd;
    data['openingTime'] = this.openingTime;
    data['closingTime'] = this.closingTime;
    return data;
  }
}
